### ISSUE TITLE
Inconsistent virtual method obfuscation state detected.

### ISSUE THREAD

**Issue by lextudio-support** [obfuscar/obfuscar#507]
Hi.

Thank you for Obfuscar 2.0 rc7.

Windows 7 x64 SP1
.NET 3.5 assembly

Configuration file:
```
<?xml version='1.0'?>
<Obfuscator>
  <Var name="KeepPublicApi" value="true" />
  <Var name="HidePrivateApi" value="true" />
  <Var name="KeyFile" value="..\..\..\StrongName.snk" />

  <Module file="..\..\..\MyAssembly\bin\Release\MyAssembly.exe" />
</Obfuscator>
```

Output:
```
...
Trying to resolve dependency: mscorlib, Version=2.0.0.0, Culture=neutral, Public
KeyToken=b77a5c561934e089
Done.
Renaming:  fields...parameters...properties...events...methods...
An error occurred during processing:
Inconsistent virtual method obfuscation state detected. Abort. Please review the
 following methods,
[MyAssembly]♫? ::♫?  ? ☻[0]->WillRename:A
[MyAssembly]♠  ::♫?  ? ☻[0]->Skipped:special name
[MyAssembly]♫  ::♫?  ? ☻[0]->WillRename:A
```

The assembly has been merged with ilmerge and it seems that there's a reference to another assembly that has already been obfuscated.

Any ideas?

Thank you.

**Original Reported Date**: 2014-06-17T12:57:55.983-07:00
**Planned For Release**: 
**Original CodePlex ID**: 16

**Comment by lextudio-support** [comment]
**Copied from CodePlex without authors**:

Did you try to obfuscate an assembly that's obfuscated by another tool? The function names are so strange, and I am not sure if that's the cause of the problem.

The second method was reported as a special named method based on the log, and such decision is mainly driven by Mono.Cecil. Without analyzing this assembly, the conclusion cannot be drawn. If you like, you can share it with me via support@lextm.com.

**Original Posted Date**: 2014-06-19T23:03:11.273-07:00

**Comment by lextudio-support** [comment]
**Copied from CodePlex without authors**:

A blog post has been published regarding this issue,

http://blog.lextudio.com/2014/09/obfuscar-the-story-behind-inconsistent-virtual-method-obfuscation-state-detected/

**Original Posted Date**: 2014-09-14T00:18:15.31-07:00

**Comment by lextudio-support** [comment]
**Original Closed Date**: 2014-10-13T00:57:40.84-07:00
